### DISCUSSION TITLE
My thoughts on the 'awful' instructions in the readme

### DISCUSSION THREAD

**Discussion by Zonnev** [Zonnev/elementaryos-firefox-theme#209] [General]
Hello everybody,

I've added support for Firefox ESR installations since @nevillepark asked me to add it and I find that installing ESR is the most easy way to have Firefox on elementary OS and the best thing is that accent colours work out of the box on it!

Now the 'endless' and 'awful' instructions in the readme about installing Firefox from a package is not making me feel right and maybe scaring people away from using Firefox on elementary OS. I want to make things easy but I'm not succeeding in it, yet.

Flatpak apps soon need to accept the accent colours of the desktop environment as a default so that I can quickly delete the instructions in After Installation and everything about creating a desktop entry. Things like installing an app and integrate it in the UI of the OS should not be so difficult like it is now. It is not good for the OS and the app in question but most of all users do not benefit from it at all.

Overall I am very happy to make a little difference in this situation by providing a theme for Firefox to integrate it but I am embarrassed about the immense instructions in the readme that I had to make. But I hope you all understand the importance why these instructions for installation with accent colours are there... I had to add these instructions because my goal is to integrate Firefox into elementary OS the best way I can and help anyone wanting to do the same!

Yesterday I read that Firefox 133 finally took the accent colours from Gnome, so hopefully Pantheon will follow soon as well so the readme can be much shorter...

**Comment by Zonnev** [comment]
I think it is not Pantheon's fault but that it has to do with flatpak Firefox not reading the portal responsible for the accent colours. I reported the issue at [Bugzilla](https://bugzilla.mozilla.org/show_bug.cgi?id=1935012).

**Comment by nevillepark** [comment]
Thanks so much! 

I actually installed the ESR version because for some reason Firefox 133+ (the .deb version, installed from the PPA the same way as Firefox ESR) is in dark mode no matter what time it is; not just the window chrome but the websites themselves if they use "prefers-color-scheme" in the CSS. Couldn't find anything effective/relevant in settings or about:config, realized it might be a GTK thing, thought about how I would go about tracking down the cause and reporting it as a bug, got mentally exhausted and gave up. Hopefully by the time FF ESR catches up, later versions of Firefox will behave differently and I can just switch back.

**Comment by Zonnev** [comment]
I think Mozilla is going to default to Flatpak instead of binary installations (packaged). My bug about Flatpak has been added (blocked) to a bigger bug about 'that Flatpak should be the default installation method instead of the not user friendly binary installations'. For me there is nothing wrong about Flatpak installs, only the accent color doesn't work, so hopefully they will do something about it.

Also the dark mode schedule on Flatpak Firefox just works, while on Firefox Nightly it doesn't work so well. I have Nightly installed from the package on their website (binary install). It follows darkmode when it's set in elementary OS Settings, but it opens in light mode when set to dark with the desktop switch (top right menu on the desktop). Very weird and probably a Firefox binary bug in combination with Pantheon.

So it seems every version of Firefox has bugs with theming but as they are going to Flatpak as default install on Linux, it's good they collect all the bugs before rolling it out to users.

**Reply by nevillepark** [reply]
Thanks, good to know!
